### PR TITLE
Feat/add exchange rate format

### PR DESCRIPTION
**Pull request by cisel**
# Description
Ticket: https://rvu.atlassian.net/browse/MONEY-789

Added the formatter to be compatible with the legacy product table

Screenshot 2021-11-23 at 16 33 26

# Checklist
Pull request contains:

- [ ] A new component
- [ ] Component maintenance: improvement / bug fix / etc
- [ ] Component library change: storybook / webpack / etc

Definition of done:

- [ ] Includes theme changes for both Uswitch and money
- [ ] Work has been tested in multiple browsers
- [ ] PR description includes description of change
- [ ] PR description includes screenshot of change
- [ ] If new component, designer has approved screenshot
- [ ] If the change will affect other teams, that team knows about this change
- [ ] If introducing a new component behaviour, added a story to cover that case.

### CODE REVIEW

File: src/compounds/product-table/src/generics.ts [hunk 1]
```diff
@@ -76,6 +78,16 @@ const formatters: { [unit: string]: (value: number) => string } = {
     if (megabytes >= 1000) return `${Math.floor(megabytes / 1000)}GB`
 
     return `${megabytes || 0}MB`
+  },
+
+  'exchange-rate': (value, count = 4) => {
```
Comment (samhstn) [inline src/compounds/product-table/src/generics.ts:83]: What is `count`? Is that the number of decimal places?

Comment (cisel) [reply src/compounds/product-table/src/generics.ts:83]: Yes, it is. @samhstn

**Review by erkinun — APPROVED** [review]

File: src/compounds/product-table/src/generics.ts [hunk 2]
```diff
@@ -76,6 +78,16 @@ const formatters: { [unit: string]: (value: number) => string } = {
...
+    const regex = new RegExp(`^-?\\d+(?:\\.\\d{0,${count}})?`)
+    const matches = value.toString().match(regex)
+    const withDecimals = matches?.length ? matches[0] : ''
+    const [before = '', after = ''] = withDecimals.split('.')
+    const val = `${before}.${after.padEnd(count, '0')}`
+
+    return val
```
Comment (samhstn) [inline src/compounds/product-table/src/generics.ts:90]: could we instead do:

```javascript
return Intl.NumberFormat('en-GB' , {maximumFractionDigits: count, minimumFractionDigits: count}).format(value)
```

Comment (cisel) [reply src/compounds/product-table/src/generics.ts:90]: `Intl.NumberFormat` rounds up when the number has more than 4 decimal places but Akanksha wanted to get the first 4 decimal places without rounding it up.

**Pull request merged**